docs/learn/evals.md

```diff
@@ -253,23 +320,351 @@ def evaluate_bias(query: str, generation: str):
+    def llm_evaluate_clarity(text: str):
+        ...
+    
+    def hardcoded_evaluate_length(text: str) -> float:
+        return min(len(text) / 500, 1.0)  # Normalize to 0-1 scale, capped at 1
+    
+    def combined_evaluation(text: str) -> dict:
+        llm_eval = llm_evaluate_clarity(text)
+        length_eval = hardcoded_evaluate_length(text)
+        
+        return {
+            "llm_clarity_score": llm_eval.score,
+            "llm_reasoning": llm_eval.reasoning,
+            "length_score": length_eval,
+            "combined_score": (llm_eval.score + length_eval) / 2
+        }
+    
+    # Usage
+    text = "This is a sample text for evaluation."
+    result = combined_evaluation(text)
+    print(result)
+    ```
+
+2. **Consistent Scaling**: When using numerical scores, ensure consistent scaling across different evaluation criteria.
+
+    ```python  hl_lines="4-7 12-13"
+    from mirascope.core import openai, prompt_template
+    from pydantic import BaseModel, Field
+    
+    class EvalScores(BaseModel):
+        clarity: float = Field(..., description="Score between 0.0 and 1.0")
+        relevance: float = Field(..., description="Score between 0.0 and 1.0")
+        coherence: float = Field(..., description="Score between 0.0 and 1.0")
+    
+    @openai.call(model="gpt-4o", response_model=EvalScores)
+    @prompt_template("""
+    Evaluate the following text on three criteria:
+    1. Clarity: How clear and easy to understand is the text?
+    2. Relevance: How relevant is the text to the given topic?
+    3. Coherence: How well does the text flow and maintain logical connections?
+    
+    Provide a score for each criterion on a scale of 0.0 to 1.0, where 0.0 is the lowest and 1.0 is the highest.
+    
+    Text: {text}
+    Topic: {topic}
+    """)
+    def evaluate_text(text: str, topic: str):
+        ...
+    
+    # Usage
+    text = "This is a sample text about artificial intelligence."
+    topic = "Artificial Intelligence"
+    scores = evaluate_text(text=text, topic=topic)
+    print(scores)
+    ```
+
+3. **Multiple Judges**: For important evaluations, consider using a panel of LLM judges and aggregate their scores.
+
+    ```python  hl_lines="22 23"
+    import asyncio
+    from mirascope.core import openai, anthropic, prompt_template
+    from pydantic import BaseModel, Field
+    from statistics import mean
+    
+    class JudgeEval(BaseModel):
+        score: float = Field(..., description="Score between 0.0 and 1.0")
+        reasoning: str
+    
+    @openai.call(model="gpt-4o", response_model=JudgeEval)
+    @prompt_template("Rate the quality of this text from 0.0 to 1.0: {text}")
+    def openai_judge(text: str):
+        ...
+    
+    @anthropic.call(model="claude-3-5-sonnet-20240620", response_model=JudgeEval)
+    @prompt_template("Rate the quality of this text from 0.0 to 1.0: {text}")
+    def anthropic_judge(text: str):
```

Comment:
these should use the same prompt template and decorate inside the function right? otherwise there is a lot of duplicated code